pycontrails/core/flight.py

```diff
@@ -900,6 +901,62 @@ def resample_and_fill(
         df = df.reset_index()
         return Flight(data=df, attrs=self.attrs)
 
+    def fit_flight_profile(
```

Comment:
To be safe, I think you should add this same method to the `Fleet` class, and it can simply raise `NotImplementedError("Only implemented for Flight instances")`